pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -33,15 +33,21 @@ func addDefaultingFuncs(scheme *kruntime.Scheme) error {
 	return RegisterDefaults(scheme)
 }
 
+func SetDefaults_CloudControllerManagerConfiguration(obj *CloudControllerManagerConfiguration) {
+	zero := metav1.Duration{}
+	if obj.ServiceController.ConcurrentServiceSyncs == 0 {
```

Comment:
I think we should apply defaults in the command, not in the API layer.  Since this is alpha and not serialized, we can change later, but something to think about moving forward.

Reply:
got it, thanks

Reply:
@deads2k this is worth a longer more broad discussion. The whole componentconfig approach is based on exactly this pattern to use the API default as the command defaults. @mtaufen has a discussion in his doc.